app/gui/src/project-view/components/GraphEditor/GraphEdge.vue

```diff
@@ -150,7 +154,13 @@ const sourceMask = computed<NodeMask | undefined>(() => {
   return { rect, radius }
 })
 
-const { baseColor, selected, pending } = useComponentColors(graph.db, selection, sourceNode)
+const {
+  baseColor,
+  selected: nodeSelected,
+  pending,
+} = useComponentColors(graph.db, selection, sourceNode)
+const edgeSelected = computed(() => shallowEquality(selection?.selectedEdge, edge))
```

Comment:
I think this comparison is somewhat fragile. `Edge` is a complex type; we may add a field to one of its concrete subtypes that cannot be correctly compared by `shallowEquality`.

If we define a function that returns a string uniquely identifying an edge, and use that for comparisons, type safety will help ensure that the comparison remains correct through future changes.